lldb/tools/lldb-dap/DAP.h

```diff
@@ -417,8 +424,11 @@ struct DAP {
   lldb::SBMutex GetAPIMutex() const { return target.GetAPIMutex(); }
 
 private:
-  std::mutex m_queue_mutex;
+  /// Queue for all incoming messages.
   std::deque<protocol::Message> m_queue;
+  /// Dedicated queue for launching and attaching.
+  std::deque<protocol::Message> m_launch_attach_queue;
```

Comment:
I think there is only ever one launch or attach, should this be an optional message instead? Also named `m_pending_launch_or_attach`?

Reply:
I changed it to an optional, but then backed it out again because we also have to delay the initial `threads` request until we've completed the launch or attach. Let me know if you prefer two optionals instead (i.e. `m_pending_launch_or_attach` and `m_pending_threads`).

Reply:
Looking at things now, I think a separate queue makes more sense.